src/redismodule.h

```diff
@@ -1333,6 +1342,18 @@ REDISMODULE_API void (*RedisModule_RdbStreamFree)(RedisModuleRdbStream *stream)
 REDISMODULE_API int (*RedisModule_RdbLoad)(RedisModuleCtx *ctx, RedisModuleRdbStream *stream, int flags) REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_RdbSave)(RedisModuleCtx *ctx, RedisModuleRdbStream *stream, int flags) REDISMODULE_ATTR;
 REDISMODULE_API const char * (*RedisModule_GetInternalSecret)(RedisModuleCtx *ctx, size_t *len) REDISMODULE_ATTR;
+REDISMODULE_API RedisModuleConfigIterator* (*RedisModule_GetConfigIterator)(RedisModuleCtx *ctx, const char *pattern) REDISMODULE_ATTR;
+REDISMODULE_API void (*RedisModule_ReleaseConfigIterator)(RedisModuleCtx *ctx, RedisModuleConfigIterator *iter) REDISMODULE_ATTR;
+REDISMODULE_API const char * (*RedisModule_ConfigIteratorNext)(RedisModuleConfigIterator *iter, RedisModuleConfigType *typehint) REDISMODULE_ATTR;
+REDISMODULE_API RedisModuleConfigType (*RedisModule_GetConfigType)(const char *name) REDISMODULE_ATTR;
```

Comment:
Changing `RedisModule_GetConfigType`'s prototype would allow it  return `REDISMODULE_ERR` on error and drop `REDISMODULE_CONFIG_TYPE_UNKNOWN` enum as @sundb proposed above. 

```
REDISMODULE_API int (*RedisModule_GetConfigType)(const char *name, RedisModuleConfigType *type) REDISMODULE_ATTR;
```

Reply:
yep it was already done in [6cf4e63](https://github.com/minchopaskal/redis/pull/6/commits/6cf4e63af2af53572277691956e9ee8e0cf95c79)